src/Microsoft.AspNet.Mvc.Core/ModelBinding/Validation/DataAnnotationsClientModelValidatorProvider.cs

```diff
@@ -32,14 +32,24 @@ internal delegate IClientModelValidator
         /// <inheritdoc />
         public void GetValidators(ClientValidatorProviderContext context)
         {
+            var hasRequiredAttribute = false;
+
             foreach (var attribute in context.ValidatorMetadata.OfType<ValidationAttribute>())
             {
+                hasRequiredAttribute |= attribute is RequiredAttribute; 
+
                 DataAnnotationsClientModelValidationFactory factory;
                 if (_attributeFactories.TryGetValue(attribute.GetType(), out factory))
                 {
                     context.Validators.Add(factory(attribute));
                 }
             }
+
+            if (!hasRequiredAttribute && context.ModelMetadata.IsRequired)
+            {
+                // Add a default '[Required]' validator for generating HTML if necessary.
+                context.Validators.Add(new RequiredAttributeAdapter(new RequiredAttribute()));
```

Comment:
Agree with offline discussion. Should also add a `RequiredAttribute` to the validations that will be run on the server.